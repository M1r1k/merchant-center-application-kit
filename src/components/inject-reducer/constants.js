// eslint-disable-next-line import/prefer-default-export
export const ACTIVATE_PLUGIN = 'ACTIVATE_PLUGIN';
