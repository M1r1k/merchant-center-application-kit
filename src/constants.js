// eslint-disable-next-line import/prefer-default-export
export const USER_PROFILE_FORM_NAME =
  'packages-shared/app-shell/USER_PROFILE_FORM_NAME';
