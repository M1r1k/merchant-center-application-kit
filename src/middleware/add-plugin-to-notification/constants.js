export const __LOCAL = 'LOCAL';
export const __GLOBAL = 'GLOBAL';
