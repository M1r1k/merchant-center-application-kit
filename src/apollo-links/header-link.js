import { ApolloLink } from 'apollo-link';
import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import { getCorrelationId, selectProjectKey } from '../utils';

const isKnownTarget = target => Object.values(GRAPHQL_TARGETS).includes(target);

/* eslint-disable import/prefer-default-export */
// Use a middleware to update the request headers with the correct params.
export const headerLink = new ApolloLink((operation, forward) => {
  const target = operation.variables.target;
  if (!isKnownTarget(target))
    throw new Error(
      `GraphQL target "${target}" is missing or is not supported`
    );

  const projectKey = selectProjectKey();
  // NOTE: keep header names with capital letters to avoid possible conflicts
  // or problems with nginx.
  operation.setContext({
    headers: {
      'X-Project-Key': projectKey,
      'X-Correlation-Id': getCorrelationId(),
      'X-Graphql-Target': target,
    },
  });
  return forward(operation);
});
