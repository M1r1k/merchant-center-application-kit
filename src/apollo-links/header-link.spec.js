import { ApolloLink, execute, Observable } from 'apollo-link';
import gql from 'graphql-tag';
import waitFor from 'wait-for-observables';
import { GRAPHQL_TARGETS } from '@commercetools-local/constants';
import { createHeaderLink } from './header-link';

const query = gql`
  {
    sample {
      id
    }
  }
`;

describe('with valid target', () => {
  let context;
  let debugLink;
  let headerLink;
  let link;
  let terminatingLinkStub;

  beforeEach(async () => {
    debugLink = new ApolloLink((operation, forward) => {
      context = operation.getContext();

      return forward(operation);
    });

    headerLink = createHeaderLink({
      storage: {
        get: jest.fn(() => 'project-1'),
      },
    });

    terminatingLinkStub = jest.fn(() => Observable.of({}));

    link = ApolloLink.from([headerLink, debugLink, terminatingLinkStub]);

    await waitFor(
      execute(link, {
        query,
        variables: { target: GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND },
      })
    );
  });

  it('should set headers matching snapshot', () => {
    expect(context).toMatchSnapshot();
  });

  it('should set `X-Project-Key`-Header', () => {
    expect(context.headers).toEqual(
      expect.objectContaining({
        'X-Project-Key': 'project-1',
      })
    );
  });

  it('should set `X-Graphql-Target`-Header', () => {
    expect(context.headers).toEqual(
      expect.objectContaining({
        'X-Graphql-Target': GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND,
      })
    );
  });
});
