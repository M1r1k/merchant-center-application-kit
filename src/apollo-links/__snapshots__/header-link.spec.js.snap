// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with valid target should set headers matching snapshot 1`] = `
Object {
  "headers": Object {
    "X-Correlation-Id": "mc/project-1/foo-uuid",
    "X-Graphql-Target": "mc",
    "X-Project-Key": "project-1",
  },
}
`;
